src/components/epic/signup/hooks/useSignupForm.ts

```diff
@@ -43,7 +51,12 @@ export const useSignupForm = (): SignupFormHook => {
     signup(data);
   });
 
+  const triggerUserNameValidation = () => {
+    trigger("userName");
+  };
```

Comment:
By default the form validates `onSubmit` only. But for the username we want to plugin an exception: we want to trigger a validation o blur. Unfortunately there is no declarative way of doing this, we have to imperatively plug a trigger to the `onBlur` of the input.